added 15112 :)

fredasu opened the pull request:
# Summary of Changes

- Please indicate the classes you are modifying:
  - made 15112

# Checklist:

Make sure that your contributions:

- [ ] Are accurate and will not mislead the reader (put in a disclaimer about being remote S21)
- [ ] Do not encourage or cause academic integrity violations
- [ ] Do not offend professors or unfairly criticize the university

Commit:
added 15112 :)

mikinty commented on cs_core/15112.md:
```diff
@@ -0,0 +1,106 @@
+
+## Resources
+
+- syllabus if you want to look at the notes/schedule: https://www.cs.cmu.edu/~112/syllabus.html
```
Do you mind formatting the link here to be markdown friendly? I think github by default recognizes it, but just to be more official 

```
[title](https://www.example.com)
```
```suggestion
- syllabus if you want to look at the notes/schedule: [https://www.cs.cmu.edu/~112/syllabus.html](https://www.cs.cmu.edu/~112/syllabus.html)
```

mikinty commented on cs_core/15112.md:
```diff
@@ -0,0 +1,106 @@
+- HW gradually get harder over time, but the difficulty will flatten out by 
+week 4 according to the teachers. IMO I think the difficulty level depends on how
+much experience you've had with the topic before.
+- Technology failures: back up your code somewhere, and don't submit for the
```
Is `git` a reasonable recommendation here for file management, or is that too complicated for 15-112? 

When I was at CMU, CS majors had to learn `git` in some class (although you barely did), but 112 is taken by so many non-CS majors, it's unlikely that most people will know what it is. 

Version control is extremely important at higher level classes, but getting a headstart in 112 could be a good idea

mikinty reviewed (changes requested):
Can you please add the link to this on the main table of contents please?

Overall, fantastic article, as it gives good insight and personal experiences into what it is like to take the class. Great for people considering taking the class or people currently in it. Thanks!

fredasu commented:
Hi, thanks for the feedback! Sorry for the late reply.

Is there an easy way to make changes or would I need to re-upload again?
I'm new to this github stuff.

Also, I don't think github was needed for 15-112 because they allowed other
forms of backup as well, such as Google Drive.

Thanks!

Freda

On Sun, Mar 20, 2022 at 8:56 PM Michael You ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Can you please add the link to this on the main table of contents please?
>
> Overall, fantastic article, as it gives good insight and personal
> experiences into what it is like to take the class. Great for people
> considering taking the class or people currently in it. Thanks!
> ------------------------------
>
> In cs_core/15112.md
> <https://github.com/CMU-HKN/CMU-ECE-CS-Guide/pull/50#discussion_r830702155>
> :
>
> > +- Don't rely on a curve. Sure they might curve a quiz a little bit in the beginning
>
> +when people are still adjusting to the course, but it's definitely not a regular thing.
>
> +- Term project counts for a good chunk of your grade so spend quality time on it.
>
> +
>
> +## What to watch out for
>
> +
>
> +- The first quiz, as mentioned above, is usually a bit rough.
>
> +- HW gradually get harder over time, but the difficulty will flatten out by
>
> +week 4 according to the teachers. IMO I think the difficulty level depends on how
>
> +much experience you've had with the topic before.
>
> +- Technology failures: back up your code somewhere, and don't submit for the
>
> +first time too close to the deadline.
>
> +
>
> +## Resources
>
> +
>
> +- syllabus if you want to look at the notes/schedule: https://www.cs.cmu.edu/~112/syllabus.html
>
> Do you mind formatting the link here to be markdown friendly? I think
> github by default recognizes it, but just to be more official
>
> [title](https://www.example.com)
>
>
> ⬇️ Suggested change
>
> -- syllabus if you want to look at the notes/schedule: https://www.cs.cmu.edu/~112/syllabus.html
>
> +- syllabus if you want to look at the notes/schedule: [https://www.cs.cmu.edu/~112/syllabus.html](https://www.cs.cmu.edu/~112/syllabus.html)
>
> ------------------------------
>
> In cs_core/15112.md
> <https://github.com/CMU-HKN/CMU-ECE-CS-Guide/pull/50#discussion_r830702391>
> :
>
> > +to discuss it, so they can give you feedback about whether it's viable.
>
> +- You have 2 grace days (limit 1 per hw) where you can submit hw up to <24hr
>
> +late for full credit. So use them wisely if you need to (they can't be used for
>
> +term project deadlines though).
>
> +- Complete the hw in time for OH help.
>
> +- Don't rely on a curve. Sure they might curve a quiz a little bit in the beginning
>
> +when people are still adjusting to the course, but it's definitely not a regular thing.
>
> +- Term project counts for a good chunk of your grade so spend quality time on it.
>
> +
>
> +## What to watch out for
>
> +
>
> +- The first quiz, as mentioned above, is usually a bit rough.
>
> +- HW gradually get harder over time, but the difficulty will flatten out by
>
> +week 4 according to the teachers. IMO I think the difficulty level depends on how
>
> +much experience you've had with the topic before.
>
> +- Technology failures: back up your code somewhere, and don't submit for the
>
>
> Is git a reasonable recommendation here for file management, or is that
> too complicated for 15-112?
>
> When I was at CMU, CS majors had to learn git in some class (although you
> barely did), but 112 is taken by so many non-CS majors, it's unlikely that
> most people will know what it is.
>
> Version control is extremely important at higher level classes, but
> getting a headstart in 112 could be a good idea
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CMU-HKN/CMU-ECE-CS-Guide/pull/50#pullrequestreview-915178592>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMP7DYWWIWMWEAQVCBXT3R3VA7CNFANCNFSM5RCUMTOA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

fredasu commented:
Should I add more information about how to back things up regarding github?
Also, I created github with a personal account a while ago, but don't check
this email as frequently anymore since I have my school email,
***@***.*** so would you mind switching to that one instead?

Thank you!

On Tue, Apr 5, 2022 at 6:56 PM Freda Su ***@***.***> wrote:

> Hi, thanks for the feedback! Sorry for the late reply.
>
> Is there an easy way to make changes or would I need to re-upload again?
> I'm new to this github stuff.
>
> Also, I don't think github was needed for 15-112 because they allowed
> other forms of backup as well, such as Google Drive.
>
> Thanks!
>
> Freda
>
> On Sun, Mar 20, 2022 at 8:56 PM Michael You ***@***.***>
> wrote:
>
>> ***@***.**** requested changes on this pull request.
>>
>> Can you please add the link to this on the main table of contents please?
>>
>> Overall, fantastic article, as it gives good insight and personal
>> experiences into what it is like to take the class. Great for people
>> considering taking the class or people currently in it. Thanks!
>> ------------------------------
>>
>> In cs_core/15112.md
>> <https://github.com/CMU-HKN/CMU-ECE-CS-Guide/pull/50#discussion_r830702155>
>> :
>>
>> > +- Don't rely on a curve. Sure they might curve a quiz a little bit in the beginning
>>
>> +when people are still adjusting to the course, but it's definitely not a regular thing.
>>
>> +- Term project counts for a good chunk of your grade so spend quality time on it.
>>
>> +
>>
>> +## What to watch out for
>>
>> +
>>
>> +- The first quiz, as mentioned above, is usually a bit rough.
>>
>> +- HW gradually get harder over time, but the difficulty will flatten out by
>>
>> +week 4 according to the teachers. IMO I think the difficulty level depends on how
>>
>> +much experience you've had with the topic before.
>>
>> +- Technology failures: back up your code somewhere, and don't submit for the
>>
>> +first time too close to the deadline.
>>
>> +
>>
>> +## Resources
>>
>> +
>>
>> +- syllabus if you want to look at the notes/schedule: https://www.cs.cmu.edu/~112/syllabus.html
>>
>> Do you mind formatting the link here to be markdown friendly? I think
>> github by default recognizes it, but just to be more official
>>
>> [title](https://www.example.com)
>>
>>
>> ⬇️ Suggested change
>>
>> -- syllabus if you want to look at the notes/schedule: https://www.cs.cmu.edu/~112/syllabus.html
>>
>> +- syllabus if you want to look at the notes/schedule: [https://www.cs.cmu.edu/~112/syllabus.html](https://www.cs.cmu.edu/~112/syllabus.html)
>>
>> ------------------------------
>>
>> In cs_core/15112.md
>> <https://github.com/CMU-HKN/CMU-ECE-CS-Guide/pull/50#discussion_r830702391>
>> :
>>
>> > +to discuss it, so they can give you feedback about whether it's viable.
>>
>> +- You have 2 grace days (limit 1 per hw) where you can submit hw up to <24hr
>>
>> +late for full credit. So use them wisely if you need to (they can't be used for
>>
>> +term project deadlines though).
>>
>> +- Complete the hw in time for OH help.
>>
>> +- Don't rely on a curve. Sure they might curve a quiz a little bit in the beginning
>>
>> +when people are still adjusting to the course, but it's definitely not a regular thing.
>>
>> +- Term project counts for a good chunk of your grade so spend quality time on it.
>>
>> +
>>
>> +## What to watch out for
>>
>> +
>>
>> +- The first quiz, as mentioned above, is usually a bit rough.
>>
>> +- HW gradually get harder over time, but the difficulty will flatten out by
>>
>> +week 4 according to the teachers. IMO I think the difficulty level depends on how
>>
>> +much experience you've had with the topic before.
>>
>> +- Technology failures: back up your code somewhere, and don't submit for the
>>
>>
>> Is git a reasonable recommendation here for file management, or is that
>> too complicated for 15-112?
>>
>> When I was at CMU, CS majors had to learn git in some class (although
>> you barely did), but 112 is taken by so many non-CS majors, it's unlikely
>> that most people will know what it is.
>>
>> Version control is extremely important at higher level classes, but
>> getting a headstart in 112 could be a good idea
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/CMU-HKN/CMU-ECE-CS-Guide/pull/50#pullrequestreview-915178592>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AMP7DYWWIWMWEAQVCBXT3R3VA7CNFANCNFSM5RCUMTOA>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>>
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

mikinty commented:
No worries about the delay.

Github sends an email based on your account preferences, so you can switch to your school email if you'd like. 

You should claim the [free GitHub pro and a bunch of other goodies](https://education.github.com/pack) with your student email.

To make changes, you can just add more commits here. I'd prefer it if you squash it so we don't have too many commits for this PR.

mikinty commented:
> Hi, thanks for the feedback! Sorry for the late reply. Is there an easy way to make changes or would I need to re-upload again? I'm new to this github stuff. Also, I don't think github was needed for 15-112 because they allowed other forms of backup as well, such as Google Drive. Thanks! Freda

Yeah so using `git` is probably a more legit way of backing things up, esp for finding stuff in history, but I agree Google Drive is honestly fine for 15-112. 122 and beyond is when I would consider using `git`, and classes like `213` pretty much force you to do it.

fredasu committed:
Update cs_core/15112.md

I'm not sure how to do this commit thing. Do I just click on "Commit changes"?

Co-authored-by: Michael You <youmichaelc@gmail.com>

mikinty reviewed (approved):
Thank you :)
